modal/_runtime/user_code_imports.py

```diff
@@ -327,22 +325,31 @@ def import_class_service(
 
         assert not function_def.use_method_name  # new "placeholder methods" should not be invoked directly!
         cls_name = parts[0]
-        cls = getattr(module, cls_name)
-
-    if isinstance(cls, modal.cls.Cls):
-        # The cls decorator is in global scope
-        _cls = synchronizer._translate_in(cls)
-        method_partials = _cls._get_partial_functions()
-        service_function: _Function = _cls._class_service_function
-        service_deps = service_function.deps(only_explicit_mounts=True)
-        active_app = service_function.app
+        cls_or_user_cls = getattr(module, cls_name)
+
+    if isinstance(cls_or_user_cls, modal.cls.Cls):
+        _cls = synchronizer._translate_in(cls_or_user_cls)
+        class_service_function: _Function = _cls._get_class_service_function()
+        service_deps = class_service_function.deps(only_explicit_mounts=True)
+        active_app = class_service_function.app
     else:
-        # Undecorated user class - find all methods
-        method_partials = _find_partial_methods_for_user_cls(cls, _PartialFunctionFlags.all())
-        service_deps = None
-        active_app = None
+        # Undecorated user class (serialized or local scope-decoration).
+        service_deps = None  # we can't infer service deps for now
+        active_app = get_active_app_fallback(function_def)
+        _client: "modal.client._Client" = synchronizer._translate_in(client)
+        _service_function: modal._functions._Function[Any, Any, Any] = modal._functions._Function._new_hydrated(
+            service_function_hydration_data.object_id,
+            _client,
+            service_function_hydration_data.function_handle_metadata,
+            is_another_app=True,  # this skips re-loading the function, which is required since it doesn't have a loader
```

Comment:
`is_another_app` is a very strangely named variable these days and I'm actually not sure if we need it anymore or if we could just let it be always `True`. 

It seems to have two use cases:
* Prevent re-loading of already hydrated objects being passed into Resolver.load
* Prevents a specific kind of exception that's raised when a function changes id on (re-)deploy?